.github/workflows/test.yml

```diff
@@ -0,0 +1,50 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+      - migrate-tests-to-actions
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+    - name: Run markdown link checker
+      uses: gaurav-nelson/github-action-markdown-link-check@0fe4911067fa322422f325b002d2038ba5602170
+      with:
+        config-file: '.github/workflows/markdown.links.config.json'
+        folder-path: 'website/'
+  test:
+    env:
+      GO111MODULE: on
+    strategy:
+      matrix:
+        go-version: [1.16.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@37335c7bb261b353407cff977110895fa0b4f7d8
+      with:
+        go-version: ${{ matrix.go-version }}
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.RO_DOCKERHUB_USER }}
+        password: ${{ secrets.RO_DOCKERHUB_TOKEN }}
```

Comment:
Thanks for writing this up for us! It seems to be working as expected. The only part I was confused about is the secrets. I couldn't find them in the org level or the repo level. But I think it's because I just don't have access to them. Is this expected?

Reply:
Ah right, you may need specific permissions to see them under 'secrets', but I added them as repo-level env vars. It uses use read only dockerhub creds associated with a service account.